src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -263,6 +263,17 @@ public static Assembly LoadFile(string path)
                 if (s_loadfile.TryGetValue(normalizedPath, out result))
                     return result;
 
+                // we cannot check for file presence on BROWSER. The files could be embedded and not physically present.
+#if !TARGET_BROWSER
+                string assemblyPath = normalizedPath;
```

Comment:
We have called `GetFullPath` above that does the normalization. I do not think we should be doing ad-hoc normalization here.

If the Mono loader has quirks around handling `\\` on Unix, it should be dealt with separately (ideally by deleting those quirks).